File: leetcode/26-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int currentIdx(0);
        for (auto i=currentIdx+1; i<n; ++i) {
            if (A[currentIdx] == A[i]) {
                continue;
            } else {
                A[++currentIdx] = A[i];
            }
        }
        return n == 0 ? 0 : currentIdx + 1;
    }
};

File: leetcode/26-remove-duplicates-from-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
  return 0 if nums.empty?
  b = 1
  current = nums[0]
  (1...nums.size).each do |i|
    if nums[i] != current
      current = nums[i]
      nums[b] = current
      b += 1
    end
  end
  b
end
